lib/domain/service/strategy/strategies/base_strategy.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:onix_flutter_bricks/domain/entity/config/config.dart';
+import 'package:onix_flutter_bricks/domain/entity/state_management/state_managemet_variant.dart';
+import 'package:onix_flutter_bricks/domain/repository/screen_repository.dart';
+import 'package:onix_flutter_bricks/domain/service/base/base_generation_service.dart';
+import 'package:onix_flutter_bricks/domain/service/file_generator_service/screen_generators/default_screen_route_generator.dart';
+import 'package:onix_flutter_bricks/domain/service/file_generator_service/screen_generators/params/default_screen_route_generator_params.dart';
+import 'package:onix_flutter_bricks/domain/service/file_generator_service/screen_generators/params/screen_generator_params.dart';
+import 'package:onix_flutter_bricks/domain/service/output_service/output_service.dart';
+import 'package:onix_flutter_bricks/domain/service/strategy/state_manager_strategy.dart';
+import 'package:onix_flutter_bricks/util/extension/output/output_message_extension.dart';
+
+class BaseStrategy implements StateManagerStrategy {
+  final BaseGenerationService<bool> _defaultScreenRouteGenerator =
+      DefaultScreenRouteGenerator();
+
+  @override
+  List<StateManagementVariant> get variants => [
+        const StatefulStateManagementVariant(),
+        const StatelessStateManagementVariant(),
+      ];
+
+  @override
+  Future<void> generate({
+    required Config config,
+    required ScreenRepository screenRepository,
+    required OutputService outputService,
```

Comment:
I would remove OutputService from the OutputService interface and replace it with void Function(String), essentially outputService.add would only be called in GenerateScreensUseCase than in every single class